### PR TITLE
[CLOUDTRUST-2202] create API for creating a new shadow user in a customer realm

### PR DESCRIPTION
**Pull request by bsoniam**

### CODE REVIEW

File: pkg/management/authorization.go [hunk 1]
```diff
@@ -464,3 +465,13 @@ func (c *authorizationComponentMW) UpdateRealmCustomConfiguration(ctx context.Co
 
 	return c.next.UpdateRealmCustomConfiguration(ctx, realmName, customConfig)
 }
+
+func (c *authorizationComponentMW) CreateShadowUser(ctx context.Context, realmName string, userID string, provider string, fedID api.FederatedIdentityRepresentation) error {
+	var action = MGMTCreateShadowUser.String()
+	var targetRealm = realmName
+	if err := c.authManager.CheckAuthorizationOnTargetRealm(ctx, action, targetRealm); err != nil {
```
Comment (harture) [inline pkg/management/authorization.go]: I think wee should check the targetUser not only targetRealm

File: configs/authorization.json [hunk 2]
```diff
@@ -161,7 +161,7 @@
           "*": {}
         }
       }
-    },
+      },
```
Comment (fperot74) [inline configs/authorization.json]: Bad JSON identation

File: pkg/management/component_test.go [hunk 3]
```diff
@@ -3217,3 +3217,55 @@ func TestUpdateRealmCustomConfiguration(t *testing.T) {
...
+
+	var managementComponent = NewComponent(mockKeycloakClient, mockEventDBModule, mockConfigurationDBModule, mockLogger)
+
+	var accessToken = "TOKEN=="
+	var username = "test"
+	var realmName = "master"
+	var userID = "41dbf4a8-32a9-4000-8c17-edc854c31231"
+	var provider = "provider"
+
+	// Create shadow user
```
Comment (fperot74) [inline pkg/management/component_test.go]: You can use t.Run(...)
(see in other tests how it works)

File: pkg/management/endpoint_test.go [hunk 4]
```diff
@@ -1084,6 +1084,51 @@ func TestUpdateRealmCustomConfigurationEndpoint(t *testing.T) {
...
+	var req = make(map[string]string)
+	req["userID"] = userID
+	req["provider"] = provider
+	req["realm"] = realm
+
+	fedID, _ := json.Marshal(api.FederatedIdentityRepresentation{Username: &username, UserID: &userID})
+	req["body"] = string(fedID)
+
+	// No error
+	{
```
Comment (fperot74) [inline pkg/management/endpoint_test.go]: You can use t.Run(...)

File: configs/authorization.json (hunk 2 above)
Comment (bsoniam) [reply configs/authorization.json]: Thanks, fixed

File: pkg/management/component_test.go (hunk 3 above)
Comment (bsoniam) [reply pkg/management/component_test.go]: fixed

File: pkg/management/endpoint_test.go (hunk 4 above)
Comment (bsoniam) [reply pkg/management/endpoint_test.go]: fixed

**Pull request merged**